### PR TITLE
docs: add documentation

### PR DESCRIPTION
**Pull request by lentidas**

### CODE REVIEW

File: eks/outputs.tf [hunk 1]
```diff
@@ -1,11 +1,14 @@
 output "id" {
-  value = module.thanos.id
+  description = "ID to pass other modules in order to refer to this module as a dependency. It takes the ID that comes from the main module and passes it along to the code that called this variant in the first place."
```
Comment (fsismondi) [inline eks/outputs.tf:2]: I would just say  "The module identifier. It may be used in depends_on of other modules to handle race conditions."
The rest is all implementation specific (user shouldnt care about this)

File: eks/outputs.tf [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+  value       = module.thanos.id
 }
 
 output "metrics_archives" {
-  value = local.metrics_archives
+  description = "Bucket configuration in an HCL structure that will be used as an output to pass on to the module `kube-prometheus-stack` to activate and then configure `thanos-sidecar`."
```
Comment (fsismondi) [inline eks/outputs.tf]: Similar comment here, we should avoid implementaiton specific information.
I would go with "Bucket configuration for `kube-prometheus-stack` and  `thanos-sidecar` "

File: eks/outputs.tf [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+  value       = local.metrics_archives
 }
 
 output "thanos_enabled" {
```
Comment (fsismondi) [inline eks/outputs.tf:11]: I dont think it is necesary to repeat "thanos", instead I would just output "enabled"
Referencing the output as `module.thanos.enabled` seems cleaner than `module.thanos.thanos_enabled`
I know this is not the scope of your PR but maybe we can already fix this?

File: eks/outputs.tf [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
-  value = module.thanos.thanos_enabled
+  description = "Flag to say that Thanos is enabled. It takes the output that comes from the main module and passes it along to the code that called this variant in the first place."
```
Comment (fsismondi) [inline eks/outputs.tf]: I would say only "Boolean indicating wether Thanos is enabled." 
I am not sure what is the interest of having this value though,but I guess that discussion is not related to your PR

File: outputs.tf [hunk 5]
```diff
@@ -1,7 +1,9 @@
 output "id" {
-  value = resource.null_resource.this.id
+  description = "ID to pass other modules in order to refer to this module as a dependency."
```
Comment (fsismondi) [inline outputs.tf:2]: "The module identifier. It may be used in depends_on of other modules to handle race conditions."

File: variables.tf [hunk 6]
```diff
@@ -3,35 +3,41 @@
 #######################
 
 variable "cluster_name" {
-  type = string
+  description = "Name given to the cluster. This value will be used to generate the URLs in order to create the ingresses to access the application."
```
Comment (fsismondi) [inline variables.tf]: Maybe we can simplify to:
"Name given to the cluster. Value used for the ingress' URL of the application."

File: variables.tf [hunk 7]
```diff
@@ -3,35 +3,41 @@
...
+  type        = string
 }
 
 variable "base_domain" {
-  type = string
+  description = "Base domain of the cluster. This value will be used to generate the URLs in order to create the ingresses to access the application."
```
Comment (fsismondi) [inline variables.tf]: "Base domain of the cluster. Value used for the ingress' URL of the application."

File: variables.tf [hunk 8]
```diff
@@ -3,35 +3,41 @@
...
-  default = "ca-issuer"
+  description = "SSL certificate issuer to use. Usually you would configure this value as `letsencrypt-staging` or `letsencrypt-prod` on your root `*.tf` files."
+  type        = string
+  default     = "ca-issuer"
 }
 
 variable "namespace" {
-  type    = string
-  default = "thanos"
+  description = "The namespace where the application's resources will reside (it will be created in case it dows not already exist)."
```
Comment (fsismondi) [inline variables.tf]: I believe using the same sentence form you used before makes the description easier to read:
"The namespace where the Kubernetes resources should be created. Namespace will be created in case it doesn't exist."

File: variables.tf [hunk 9]
```diff
@@ -3,35 +3,41 @@
...
+  type        = string
+  default     = "thanos"
 }
 
 variable "helm_values" {
-  description = "Helm values, passed as a list of HCL structures."
+  description = "Helm values, passed as a list of HCL structures. These values are concatenated with the default ones and then passed to the application's charts."
```
Comment (fsismondi) [inline variables.tf]: "Helm chart value overrides. They should be passed as a list of HCL structures.` maybe ?

File: variables.tf [hunk 10]
```diff
@@ -41,7 +47,7 @@ variable "dependency_ids" {
 #######################
 
 variable "thanos" {
-  description = "Thanos settings"
+  description = "Most frequently used Thanos settings. This variable is merged with the local value `thanos_defaults`, which contains some sensible defaults. You can check the default values on the link:./local.tf[`local.tf`] file. If there still is anything other that needs to be customized, you can always pass on configuration values using the variable `helm_values`."
```
Comment (fsismondi) [inline variables.tf:50]: What does "Most frequently" means here?
Do we really need this considering we have `helm_values` to pass values ? 
If this is a super set of Helm values maybe we can explode the map into different module variables.

**Review by fsismondi — CHANGES_REQUESTED** [review]

File: eks/outputs.tf (hunk 3 above)
Comment (lentidas) [reply eks/outputs.tf:11]: We did similar changes in the `kube-prometheus-stack`. In that module, since there are multiple components, we had to distinguish them by their name, like `prometheus_enabled` or `grafana_enabled`. Although it is redundant in this module's case because we would have `module.thanos.thanos_enabled`, I thought it would be better to follow the same naming convention of the other module. 

And the other issue I see with naming the output simply `enabled` is that in the event that the user calls the module something else rather than `thanos` on their DevOps Stack module's instantiations, then it would be difficult to know what this output is referencing, even though this change on the user's part would not be recommended...

**Review by fsismondi — APPROVED** [review]
